app/views/games/show.html.erb

```diff
@@ -0,0 +1,26 @@
+<body>
+ <div class="container">
+ <table class="block-center">
+     <% 4.times do |row| %>
+      <tr class="even">
+       <% 8.times do |even_cell| %>
+       <% if @pieces.where(file: even_cell + 1, rank: row * 2 + 1)[0]%>
+        <td><font class="unicode"><%= @pieces.where(file: even_cell + 1, rank: row * 2 + 1)[0].image.html_safe %></font></td> 
+        <% else %>
+        <td></td> 
+        <% end %>   
+       <% end %>
+      </tr>
+      <tr class="odd">
+       <% 8.times do |odd_cell| %>
+        <% if @pieces.where(file: odd_cell + 1, rank: row * 2 + 2)[0]%>
```

Comment:
You're still using the `where` method, which will trigger all those DB queries I mentioned. Now that you have all the pieces associated with the game stored in memory as `@pieces`, the way to avoid that is to use the `select` method instead: http://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-select